daffodil-core/src/main/scala/org/apache/daffodil/dpath/Conversions.scala

```diff
@@ -270,10 +268,10 @@ object Conversion {
       case (_, Exists) => Nil
       case (_, other) => {
         val (relevantExpr: Expression, detailMsg: String) =
-          polymorphicExpressionDiagnostics(st, tt, context)
+          context.polymorphicExpressionErrorDetails()
```

Comment:
I thought you said that the contents of this function that do polymorphic work never get hit when called in this context? Which makes sense to me. The DownStep is really the only place where we need to detect polymorphic expressions. If there is one, ``DownStep`` will create an SDE and this function will never be hit. So in this usage, this function will just always return context and an error string of "", which I'm not sure gains us anything, aside from potential confusion and extra complications.

I'd suggest just deleting this usage. If we're concerned that there does need to be some polymorphic check here, we could maybe add a function and do something like ``Assert.invariant(!isPolymorhpic)``, but I'm not sure if that's even necessary, as polymorphism isn't even relevant in this function. All this function does it get a source type and a target type and figures out how to convert between one and another. That fact that something is polymorphic is irrelevant to that decision.

Reply:
Done.